test/update_test.dart

```diff
@@ -876,6 +881,46 @@ d: 4
           'a': {'key': {}}
         });
       });
+
+      test('adds and preserves key order (ascending)', () {
+        final doc = YamlEditor('''
+a: 1
+b: 2
+c: 3
+
+
+''');
+
+        doc.update(['d'], 4);
+        expect(doc.toString(), equals('''
+a: 1
+b: 2
+c: 3
+d: 4
+
+
+'''));
+      });
+
+      test('adds at the end when no key order is present', () {
+        final doc = YamlEditor('''
+a: 1
+c: 2
+b: 3
+
+
+''');
+
+        doc.update(['d'], 4);
+        expect(doc.toString(), equals('''
+a: 1
+c: 2
+b: 3
+d: 4
+
+
+'''));
+      });
```

Comment:
These tests are awesome, have you considered add some tests in `test/testdata/input/`?

They are less powerful than this kind of testing, but it's much easier to write a lot of test cases and therefore easier to make sure that more cases are covered.

Reply:
Should I move them?

Reply:
I'd rather suggest making more or different tests. Also fine not to.